### ISSUE TITLE
Floating point exception on torch.lcm

### ISSUE THREAD

**Issue by drewshark** [pytorch/pytorch#121343]
### 🐛 Describe the bug

Here is the code triggering this issue:

```
input = torch.tensor([-9223372036854775808], dtype=torch.int64)
other = torch.tensor(215, dtype=torch.uint8)
out = torch.lcm(input,other)
```
And the torch.lcm will throw a floating point exception error. 
I am guessing that this issue may be similar to https://github.com/pytorch/pytorch/pull/120624, where an undefined behavior occurs when calculating `INT64_MIN % -1`

### Versions

[pip3] numpy==1.25.2
[pip3] torch==2.2.1
[pip3] torchaudio==2.1.0+cu121
[pip3] torchdata==0.7.0
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.16.0
[pip3] torchvision==0.16.0+cu121
[pip3] triton==2.1.0
[conda] Could not collect

**Comment by tringwald** [comment]
You're correct, `torch.lcm` internally calls `a % b` at some point, which is pretty much the same issue as #120597.
